### PR TITLE
Refactor JSON support into a separate Gradle module and document.

### PR DESCRIPTION
**Pull request by mikehearn**
JSON/Jackson support is helpful for devs in all kinds of projects, not just web servers (and we can now advertise it as a feature added in M10). For instance I need this in the command line shell.

### CODE REVIEW

File: client/jackson/build.gradle [hunk 1]
```diff
@@ -0,0 +1,36 @@
+group 'net.corda'
```
Comment (Clintonio) [inline client/jackson/build.gradle]: Is this intended to be be built independently? If not, the group, version and buildscript can all be removed.

File: settings.gradle [hunk 2]
```diff
@@ -22,4 +22,5 @@ include 'samples:network-visualiser'
 include 'samples:simm-valuation-demo'
 include 'samples:raft-notary-demo'
 include 'samples:bank-of-corda-demo'
-
+include 'client:jackson'
+findProject(':client:jackson')?.name = 'jackson'
```
Comment (Clintonio) [inline settings.gradle]: What's the purpose of the rename here? You could push this into the module itself.

**Comment by mikehearn** [comment]
@Clintonio Changes addressed, and I noticed I needed to move the unit tests too. How does it look now?

File: client/jackson/build.gradle [hunk 3]
```diff
@@ -0,0 +1,18 @@
+apply plugin: 'java'
+apply plugin: 'kotlin'
+
+repositories {
```
Comment (Clintonio) [inline client/jackson/build.gradle]: An aside but I may just add an allprojects { } with this block in it soon in the root build.gradle.

**Pull request merged**